feature: Добавленны модели <Резюме> и сопутствующие модели <Опыт рабо…

ivan-hedgehog opened the pull request:
…ты> <Информация об образовании>,добавлена регистрация этих моделей а админ, добавлен в сеттинг в INSTALLED_APPS - 'multiselectfield'(не забываем установить командой pip install django-multiselectfield), добавлена в зависимости запись о 'multiselectfield', добавлен файл константами перебора choices

ivan-hedgehog committed:
feature: Добавленны модели <Резюме> и сопутствующие модели <Опыт работы> <Информация об образовании>,добавлена регистрация этих моделей а админ, добавлен в сеттинг в INSTALLED_APPS - 'multiselectfield'(не забываем установить командой pip install django-multiselectfield), добавлена в зависимости запись о 'multiselectfield', добавлен файл константами перебора choices

margoloko commented on crm_backend/recruitment/models.py:
```diff
@@ -1,3 +1,158 @@
+    '''Опыт работы.'''
+
+    dateStart = models.DateField(verbose_name='Дата устроения')
+    dateEnd = models.DateField(
```
можно добавить валидатор, который будет проверять, что дата увольнения не может быть раньше даты выхода на работу

margoloko commented on crm_backend/recruitment/models.py:
```diff
@@ -1,3 +1,158 @@
+class Work_experience(models.Model):
+    '''Опыт работы.'''
+
+    dateStart = models.DateField(verbose_name='Дата устроения')
```
Имена переменных рекомендуют писать в формате snake_case, а не CamelCase. 
date_start
date_end

margoloko commented on crm_backend/recruitment/models.py:
```diff
@@ -1,3 +1,158 @@
 
-# Create your models here.
+
+class Work_experience(models.Model):
```
CamelCase обычно используется для названий классов
WorkExperience

DOSuzer replied:
Нужно добавить поле user, т.к. опыт работы относится к конкретному соискателю. Плюс у человека может быть несколько предыдущих работ. И добавь related_name чтобы проще было получать весь опыт пользователя.

margoloko commented on crm_backend/recruitment/models.py:
```diff
@@ -1,3 +1,158 @@
+        return self.specialization
+
+
+class Applicant_resume(models.Model):
```
тоже CamelCase

DOSuzer commented on crm_backend/recruitment/models.py:
```diff
@@ -1,3 +1,158 @@
+        return f'{self.position} - {self.organization}'
+
+
+class Education(models.Model):
```
Тоже нужно добавить связь с пользователем я думаю

DOSuzer commented on crm_backend/recruitment/models.py:
```diff
@@ -1,3 +1,158 @@
+    '''Резюме'''
+
+    applicant = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
```
если модель user берешь из settings, тогда удали импорт с 8 строки. надо оставить либо то либо то, чтобы однотипно было

ivan-hedgehog committed:
feature: Добавленны модели <Резюме> и сопутствующие модели <Опыт работы>
<Информация об образовании>,добавлена регистрация этих моделей а админ,
добавлен в сеттинг в INSTALLED_APPS - 'multiselectfield'(не забываем
установить командой pip install django-multiselectfield), добавлена в
зависимости запись о 'multiselectfield', добавлен файл константами
перебора choices"with '#' will be ignored, and an empty message aborts the commit.

ivan-hedgehog committed:
feature: удалил инпут юсера

margoloko commented on crm_backend/recruitment/models.py:
```diff
@@ -1,99 +1,80 @@
 
-class Work_experience(models.Model):
-    '''Опыт работы.'''
+class Work_Experience(models.Model):
```
CamelCase

margoloko commented on crm_backend/recruitment/models.py:
```diff
@@ -1,99 +1,80 @@
-
-class Applicant_resume(models.Model):
-    '''Резюме'''
+class Applicant_Resume(models.Model):
```
CamelCase

margoloko commented on crm_backend/recruitment/admin.py:
```diff
@@ -1,16 +1,15 @@
-@admin.register(Applicant_resume)
-class Applicant_resumeAdmin(admin.ModelAdmin):
+@admin.register(Applicant_Resume)
+class Applicant_ResumeAdmin(admin.ModelAdmin):
```
CamelCase

ivan-hedgehog committed:
bug fix: исправил наиме6нование классов

ivan-hedgehog committed:
Merge branch 'develop' into feat/model_resume

ivan-hedgehog committed:
Merge branch 'develop' into feat/model_resume
